src/Nethermind/Nethermind.Evm.Test/Eip6780Tests.cs

```diff
@@ -0,0 +1,151 @@
+            Block block = Build.A.Block.WithNumber(blockNumber)
+                .WithTimestamp(timestamp)
+                .WithTransactions(initTx, tx1, tx2).WithGasLimit(2 * gasLimit).TestObject;
+
+            ParityLikeTxTracer initTracer = new(block, initTx, ParityTraceTypes.Trace | ParityTraceTypes.StateDiff);
+            _processor.Execute(initTx, block.Header, initTracer);
+            AssertStorage(new StorageCell(contractAddress, 1), 0);
+            TestState.GetBalance(contractAddress).Should().Be(99.Ether());
+
+            ParityLikeTxTracer tracer1 = new(block, tx1, ParityTraceTypes.Trace | ParityTraceTypes.StateDiff);
+            _processor.Execute(tx1, block.Header, tracer1);
+            AssertStorage(new StorageCell(contractAddress, 1), 1);
+
+            ParityLikeTxTracer tracer2 = new(block, tx2, ParityTraceTypes.Trace | ParityTraceTypes.StateDiff);
+            _processor.Execute(tx2, block.Header, tracer2);
+
+            uint expected = onlyOnSameTransaction ? 1u : 0u;
+            AssertStorage(new StorageCell(contractAddress, 1), expected);
+
+            TestState.GetBalance(contractAddress).Should().Be(0);
+            TestState.GetBalance(TestItem.PrivateKeyB.Address).Should().Be(99.Ether());
+        }
+
+        [TestCase(MainnetSpecProvider.GrayGlacierBlockNumber, MainnetSpecProvider.CancunBlockTimestamp)]
+        public void self_destruct_in_same_transaction(long blockNumber, ulong timestamp)
+        {
+            TestState.CreateAccount(TestItem.PrivateKeyA.Address, 1000.Ether());
+            TestState.Commit(SpecProvider.GenesisSpec);
+            TestState.CommitTree(0);
+            byte[] contractCode = Prepare.EvmCode
+                        .PushData(1)
+                        .Op(Instruction.SLOAD)
+                        .PushData(1)
+                        .Op(Instruction.EQ)
+                        .PushData(17)
+                        .Op(Instruction.JUMPI)
+                        .PushData(1)
+                        .PushData(1)
+                        .Op(Instruction.SSTORE)
+                        .PushData(40)
+                        .Op(Instruction.JUMP)
+                        .Op(Instruction.JUMPDEST)
+                        .PushData(TestItem.PrivateKeyB.Address)
+                        .Op(Instruction.SELFDESTRUCT)
+                        .Op(Instruction.JUMPDEST)
+                        .Done;
+            byte[] initCode = Prepare.EvmCode.ForInitOf(contractCode).Done;
+            byte[] salt = new UInt256(123).ToBigEndian();
+            Address createTxAddress = ContractAddress.From(TestItem.PrivateKeyA.Address, 0);
+            Address contractAddress = ContractAddress.From(createTxAddress, salt, initCode);
+            byte[] tx1 = Prepare.EvmCode
+                .Create2(initCode, salt, 99.Ether())
+                .Call(contractAddress, 100000)
```

Comment:
maybe we can use `CREATE2`'s return value (it returns the destination address here), by transforming this block into this : 
```
byte[] tx1 = Prepare.EvmCode
      .Create2(initCode, salt, 99.Ether())
      .DUPx(0)
      .CALL(100000)
      .CALL(100000)
      .STOP()
      .Done;
```

Reply:
Maybe this code is more optimized but less readable. The changes do not contribute to the test besides gas optimization, maybe? which we dont care about here?